FirebaseAppDistribution/Sources/FIRAppDistribution.m

```diff
@@ -0,0 +1,291 @@
+
+                                                                          completion(error);
+                                                                        }];
+                                 }];
+}
+
+- (void)signOutTester {
+  self.authState = nil;
+  _isTesterSignedIn = false;
+}
+
+- (void)fetchReleases:(FIRAppDistributionUpdateCheckCompletion)completion {
+  NSLog(@"Token: %@", self.authState.lastTokenResponse.accessToken);
+  NSURLSession *URLSession = [NSURLSession sharedSession];
+  NSMutableURLRequest *request = [[NSMutableURLRequest alloc] init];
+  NSString *urlString =
+      [NSString stringWithFormat:ReleasesEndpointURL, [[FIRApp defaultApp] options].googleAppID];
+  [request setURL:[NSURL URLWithString:urlString]];
+  [request setHTTPMethod:@"GET"];
+  [request setValue:[NSString
+                        stringWithFormat:@"Bearer %@", self.authState.lastTokenResponse.accessToken]
+      forHTTPHeaderField:@"Authorization"];
+
+  NSURLSessionDataTask *listReleasesDataTask =
+      [URLSession dataTaskWithRequest:request
+                    completionHandler:^(NSData *data, NSURLResponse *response, NSError *error) {
+                      if (error) {
+                        // TODO: Reformat error into error code
+                        completion(nil, error);
+                        return;
+                      }
+
+                      NSHTTPURLResponse *httpResponse = (NSHTTPURLResponse *)response;
+
+                      if (httpResponse.statusCode == 200) {
+                        [self handleReleasesAPIResponseWithData:data completion:completion];
+                      } else {
+                        NSLog(@"Error Response Code: %ld", httpResponse.statusCode);
+
+                        // TODO: Handle non-200 http response
+                        dispatch_async(dispatch_get_main_queue(), ^{
+                          completion(nil, nil);
+                        });
+                      }
+                    }];
+
+  [listReleasesDataTask resume];
+}
+
+- (void)handleReleasesAPIResponseWithData:data
+                               completion:(FIRAppDistributionUpdateCheckCompletion)completion {
+  // TODO: Parse response from tester API, check instance identifier and maybe return a release
+
+  // Ensure we dispatch on the main thread to allow any UI to update
+  dispatch_async(dispatch_get_main_queue(), ^{
+    completion(nil, nil);
+  });
+}
+
+- (void)checkForUpdateWithCompletion:(FIRAppDistributionUpdateCheckCompletion)completion {
+  if (self.isTesterSignedIn) {
+    [self fetchReleases:completion];
+  } else {
+    UIAlertController *alert = [UIAlertController
+        alertControllerWithTitle:@"Enable in-app alerts"
+                         message:@"Sign in with your Firebase App Distribution Google account to "
+                                 @"turn on in-app alerts for new test releases."
```

Comment:
Hmm. That's a good question. We don't have plans for localization yet. But I can see this being localized in the future. How do other SDK's do localized strings?

Reply:
Very few SDKs surface user-facing errors, only developer facing errors. cc @christibbs who runs the FIAM SDK which shows a user interface.

Reply:
We currently don't localize, but apparently this will be a build warning (see #5255 ). I'm planning on just using the localized string macro without localizing-- but that's because in our SDK it's such a rare occurrence to show this alert that it's probably not worth the effort.